flink-runtime/src/main/java/org/apache/flink/runtime/rest/FileUploadHandler.java

```diff
@@ -151,18 +143,26 @@ protected void channelRead0(final ChannelHandlerContext ctx, final HttpObject ms
 				ctx.fireChannelRead(msg);
 			}
 		} catch (Exception e) {
-			HttpRequest tmpRequest = currentHttpRequest;
-			deleteUploadedFiles();
-			reset();
-			LOG.warn("Internal server error. File upload failed.", e);
-			HandlerUtils.sendErrorResponse(
-				ctx,
-				tmpRequest,
-				new ErrorResponseBody("File upload failed."),
-				HttpResponseStatus.INTERNAL_SERVER_ERROR,
-				Collections.emptyMap()
-			);
+			handleError(ctx, "File upload failed.", HttpResponseStatus.INTERNAL_SERVER_ERROR, e);
+		}
+	}
+
+	private void handleError(ChannelHandlerContext ctx, String errorMessage, HttpResponseStatus responseStatus, @Nullable Throwable e) {
+		HttpRequest tmpRequest = currentHttpRequest;
+		deleteUploadedFiles();
+		reset();
+		if (e == null) {
+			LOG.warn(errorMessage);
+		} else {
+			LOG.warn(errorMessage, e);
```

Comment:
I think we don't have to make this distinction here. `LOG.warn(errorMessage, null)` should also work.

Reply:
but this will also print "null", won't it? That's what i was trying to avoid here.

Reply:
I don't think so.